content/dev/plugins/help.md

```diff
@@ -0,0 +1,111 @@
+
+## Help for Modules (Shipping, Payment or Order Totals) 
+
+TBD - waiting on PR 4246 to be merged. 
+
+## Examples of PHP Help Files 
+
+Quite often plain text help (i.e. a `.html` file) is perfectly adequate, but if you need to make reference to the values of settings, pull in files, or do other dynamic operations, you'll want to use a `.php` help file.
+
+### Setup 
+To have access to database settings, functions, and everything else, you need to pull in the configure file.  This is a two step process since you're not at the top level of either admin or the storefront.  First, `chdir` to the admin (or catalog) folder, then pull it in.
+
+```
+<?php 
+chdir("..");
+require 'includes/application_top.php';
+?>
+```
+
+You still have to account for being in a subfolder when building links - see examples below. 
+
+### Displaying and Linking to Config values in Core menus 
+
+Suppose you created a database entry with configuration key SPECIAL_FEE in the Admin > Configuration > My Store section.  Here's how you could access it and reference it: 
+
+```
+<?php
+echo 'Current Special Fee is $' . SPECIAL_FEE . '.<br><br>';
+echo 'To change this value, go to <a href="../configuration.php?gID=1" target="_blank">Admin > Configuration > My Store</a> <br>';
+?>
+```
+
+### Displaying and Linking to Config values in Plugin menus 
+
+If you don't know what the group id is, you have to query it. 
+Rather than querying the group title (a text string that could change), query the configuration group of a specific key in that group. 
+
+```
+<?php 
+$group = $db->Execute("SELECT * FROM " . TABLE_CONFIGURATION . " WHERE configuration_key = 'PO_SEND_PACKING_LISTS'");
+$gID = $group->fields['configuration_group_id'];
+echo 'Settings for this plugin are configured in <a href="'  . HTTPS_CATALOG_SERVER . DIR_WS_ADMIN . '../configuration.php?gID=' . $gID . '" target="_blank">Admin > Configuration > Dropship Purchase Orders</a> <br>';
+?>
+```
+
+### Pulling in a define page 
+
+Using a custom built [define page](/user/template/define_pages/) to store content means the storeowner can update the help independently without needing a developer.
+
+Here's how to pull it into a custom PHP help page. The page should be in `includes/languages/english/define_myplugin.php`.
+
+```
+<?php
+   echo 'Note: To edit this file, go to <a href="'  . HTTPS_CATALOG_SERVER . DIR_WS_ADMIN . '../define_pages_editor.php" target="_blank">Admin > Tools > Define Pages Editor</a> and select "define_myplugin.php"<br><br>';
+   $define_page = zen_get_file_directory(DIR_FS_CATALOG_LANGUAGES . $_SESSION['language'] . '/html_includes/', 'define_myplugin', 'false');
+   $text = file_get_contents($define_page);
+   echo $text;
+?>
+```
+
+### Pulling in a function 
+
+If you have created a function with a bunch of hardcoded cases that the storeowner needs to refer to (perhaps to tell you when to add a new one), you can use reflection to show the code. 
+
+```
```

Comment:
It should be explained that this is highly advanced stuff.

Reply:
Do you want to standardize on using "Technical - " as the first word in the title like you did for the UTF conversion article?  I thought that was a nice way to handle it.

Reply:
I have no strong preference at this moment. Just something to make them think twice instead of expecting it to be a walk in the park ... and expectation that it may be hard to obtain support if they stray from the pure example shown and find theirs not working ...... without saying all that!, lol